src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -0,0 +1,205 @@
+    appender.setContext(ctx)
+    appender.start()
+
+    val fileLogger = ctx.getLogger(name)
+    fileLogger.addAppender(appender)
+    fileLogger.setAdditive(false)
+    fileLogger.setLevel(level)
+    fileLogger
+  }
+
+  private def logWriteDisabled() = workflowLogger.warn(writeDisabled)
+  private def logReadDisabled() = workflowLogger.warn(readDisabled)
+}
+
+object WorkflowDescriptor {
+  def apply(id: WorkflowId, sourceFiles: WorkflowSourceFiles): WorkflowDescriptor = {
+    WorkflowDescriptor(id, sourceFiles, ConfigFactory.load)
+  }
+
+  def apply(id: WorkflowId, sourceFiles: WorkflowSourceFiles, conf: Config): WorkflowDescriptor = {
+    validateWorkflowDescriptor(id, sourceFiles, CromwellServer.backend, conf) match {
+      case scalaz.Success(w) => w
+      case scalaz.Failure(f) => throw new IllegalArgumentException(s"""\n${f.list.mkString("\n")}""")
+    }
+  }
+
+  private def validateWorkflowDescriptor(id: WorkflowId,
+                                         sourceFiles: WorkflowSourceFiles,
+                                         backend: Backend,
+                                         conf: Config): ErrorOr[WorkflowDescriptor] = {
+    val namespace = validateNamespace(sourceFiles.wdlSource, backend)
+    val rawInputs = validateRawInputs(id, sourceFiles.inputsJson)
+    val options = validateWorkflowOptions(id, sourceFiles.workflowOptionsJson, backend)
+    (namespace |@| rawInputs |@| options) { (_, _, _) } match {
+      case scalaz.Success((n, r, o)) =>
+        val validatedDescriptor = for {
+          c <- validateCoercedInputs(id, r, n).disjunction
+          d <- validateDeclarations(id, n, o, c, backend).disjunction
+        } yield WorkflowDescriptor(id, sourceFiles, o, r, n, c, d, backend, configCallCaching(conf), lookupDockerHash(conf))
+        validatedDescriptor.validation
+      case scalaz.Failure(f) => f.list.mkString("").failureNel
+    }
+  }
+
+  private def validateNamespace(source: WdlSource, backend: Backend): ErrorOr[NamespaceWithWorkflow] = {
+    try {
+      NamespaceWithWorkflow.load(source, backend.backendType).successNel
+    } catch {
+      case e: Exception => s"Workflow ${id.toString} unable to load namespace: ${e.getLocalizedMessage}".failureNel
+    }
+  }
+
+  private def validateWorkflowOptions(id: WorkflowId,
+                                      optionsJson: WorkflowOptionsJson,
+                                      backend: Backend): ErrorOr[WorkflowOptions] = {
+    WorkflowOptions.fromJsonString(optionsJson) match {
+      case Success(o) => buildValidateOptions(o, id, backend)
+      case Failure(e) => s"Workflow ${id.toString} contains bad options JSON: ${e.getLocalizedMessage}".failureNel
+    }
+  }
+
+  private def buildValidateOptions(options: WorkflowOptions, id: WorkflowId, backend: Backend): ErrorOr[WorkflowOptions] = {
+    val validatedOptions = options.successNel[String]
+    val backendOptionsDisjunction = for {
+      o <- validatedOptions.disjunction
```

Comment:
Pretty sure that evolved a bit and I didn't notice I'd perturbed the thing to the extent I had